src/Microsoft.Framework.Runtime.Roslyn/RoslynCompiler.cs

```diff
@@ -186,6 +189,29 @@ public class RoslynCompiler
             return compilationContext;
         }
 
+        private CSharpCompilation ApplyStrongNameSettings(CSharpCompilation compilation)
+        {
+#if DNX451
+            // This is temporary, eventually we'll want a project.json feature for this
+            var keyFile = Environment.GetEnvironmentVariable(EnvironmentNames.BuildKeyFile);
+            if(!string.IsNullOrEmpty(keyFile))
+            {
+                var delaySignString = Environment.GetEnvironmentVariable(EnvironmentNames.BuildDelaySign);
+                var delaySign = !string.IsNullOrEmpty(delaySignString) && (
+                    string.Equals(delaySignString, "true", StringComparison.OrdinalIgnoreCase) ||
+                    string.Equals(delaySignString, "1", StringComparison.OrdinalIgnoreCase));
+
+                var strongNameProvider = new DesktopStrongNameProvider();
+                var newOptions = compilation.Options
+                    .WithStrongNameProvider(strongNameProvider)
+                    .WithCryptoKeyFile(keyFile)
+                    .WithDelaySign(!string.IsNullOrEmpty(delaySignString) && string.Equals("true", delaySignString, StringComparison.OrdinalIgnoreCase));
+                return compilation.WithOptions(newOptions);
+            }
+#endif
```

Comment:
It's not just inconsistent, it's incorrect, since the delaySign value computed above is completely unused! Going to have to see why the compiler warning didn't fire for that. Anyway, cleaning this and @Eilon 's comment in an upcoming PR. This was a quick hack-job to unblock other work :).